### DISCUSSION TITLE
Query params from window.location.href

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#6388] [Help]
I have a page called `/authorize` with a button. When I click the button, I go to Twitter and after authorization, it redirects to the callback.

The callback url shows up as:

`https://my.web/app_dev_3b368437f8274f629048339aa8dd0455?oauth_token=u8QAsgAAASDFSDFAABgYBLwJo&oauth_verifier=aWooacJu85JAdsfsdffsffdsdfsAQyjJ#/authorize`

instead of

`https://my.web/app_dev_3b368437f8274f629048339aa8dd0455#/authorize?oauth_token=u8QAsgAAASDFSDFAABgYBLwJo&oauth_verifier=aWooacJu85JAdsfsdffsffdsdfsAQyjJ`

I'd like to ideally save those both tokens to my user row.

OR

I can have another "Are you sure" button, on clicking I want to access the two params `oauth_token` and `oauth_verifier` and write them to the database.

I am not able to access `window.location.href` in my button action.

Please advise how to go about getting both the params using javascript.

**Comment by aptkingston** [comment]
Hi @SantoshSrinivas79. We deliberately sandbox JS bindings for safety, and this includes things like accessing the current window object - so there's no way to do this with a JS binding.

What you could do however, is create a new screen in your app specifically for this callback. You could create `/authorize/:oauth_token/:oauth_verifier`, and then as long as you can configure the callback URL to use URL params rather than query params, you'll be able to access these variables with URL bindings.

If you aren't able to configure your callback URL then there's no way to do this at the moment.

One feature I could suggest would be a binding in Budibase for 'Query params'. This would just be an object and you would have to know the names of your query params, but it would enable features like this.

You would probably also need another feature that's been requested before - to be able to run actions on page load.

I think I can summarise this to 2 different feature requests:
- Add binding for query params
- Add " on screen load" action

**Reply by SantoshSrinivas79** [reply]
Thanks @aptkingston .... I used a combination of `embed` as mentioned [here](https://github.com/Budibase/budibase/discussions/6377)  and the trick above.

I have a button on the second screen which could be better handled with "on screen load" action. Would be great if the two feature requests are in the product pipeline.

Any idea why the timeout trigger doesn't work on [that issue](https://github.com/Budibase/budibase/discussions/6377)? Doesn't work for me too, so I had to try a different `onhover` trigger. cc: @melohagan
